packages/compiler-cli/ngcc/src/analysis/module_with_providers_analyzer.ts

```diff
@@ -137,11 +137,15 @@ export class ModuleWithProvidersAnalyzer {
       return null;
     }
 
-    // Get hold of the return statement expression for the function
     const body = definition.body;
+    if (body === null || body.length === 0) {
+      return null;
+    }
+
+    // Get hold of the return statement expression for the function
     const lastStatement = body && body[body.length - 1];
```

Comment:
Couldn't this now be:

```suggestion
    const lastStatement = body[body.length - 1];
```